### PR TITLE
MBS-4960 / MBS-12361: Allow navigation from donation check page and improve error handling

### PR DESCRIPTION
**Pull request by reosarevok**
### Implement MBS-4960 / MBS-12361

The donation check page can be reached as a tab from your own user page, but didn't itself contain the user header and tab navigation, so the user couldn't continue navigating elsewhere from it as expected. Having the header doesn't add much complexity, so we might as well.

We were just detaching to 500 with no info if hitting the MeB donations endpoint failed. It seems a lot more sensible to give the user a proper warning instead.

### CODE REVIEW

File: root/account/Donation.js [hunk 1]
```diff
@@ -9,55 +9,76 @@
...
+  user,
+}: Props): React.Element<typeof UserAccountLayout> => (
+  <UserAccountLayout
+    entity={sanitizedAccountLayoutUser(user)}
+    page="donation"
+    title={l('Donation Check')}
+  >
+    <h2>{l('Donation Check')}</h2>
+    {checkFailed ? (
+      <Warning
```
Comment (mwiencek) [inline root/account/Donation.js:38]: This is more like an error than a warning, but not sure we have anything better at the moment.

**Review by mwiencek — APPROVED** [review]
Tested by modifying `donation_check` locally.

File: root/account/Donation.js (hunk 1 above)
Comment (reosarevok) [reply root/account/Donation.js:38]: It just felt better than a 500 with a message :)

**Comment by reosarevok** [comment]
@brainzbot, retest this please

**Pull request merged**